### DISCUSSION TITLE
Using k-anonymity for the ip address hashes

### DISCUSSION THREAD

**Discussion by daveajones** [skymethod/op3#4] [Ideas]
If a k-anonymity approach is used for the IP address hashing, it could allow for more granular regional awareness without compromising the privacy model:

https://www.immuta.com/blog/k-anonymity-everything-you-need-to-know-2021-guide/

I can see this as being a criticism (from advertisers) of the regional Cloudflare Edge CDN approach - that you are limited to only the DC's that are nearest to you.  For me, that's Atlanta, GA although I live in Birmingham, AL.  Grouping the data into geo-IP clusters and hashing against that would overcome this.

Just an idea as things progress.

If I can come up with some example code I will post here.

**Comment by johnspurlock-skymethod** [comment]
Cloudflare actually sends much more granular information in each request - right now I am intentionally dropping most of it on the floor.

Would love everyone's thoughts on which attributes we should store going forward over in discussion https://github.com/skymethod/op3/discussions/7
